docs/content/framework/interfaces/DerivativeMaterialInterface.md

```diff
@@ -0,0 +1,120 @@
+# DerivativeMaterialInterface
+
+In model development, in particular using the finite element method, it is often convenient to encapsulate expression values _together with their derivatives_. The derivatives are often needed to compute Jacobian matrix entries.
+
+In MOOSE we use material properties to provide values for expressions that can be directly computed from variables in the solve. To store the derivatives of these material properties with respect to the variables they depend on we also use material properties. We use a standardized naming scheme to name the material property derivatives. The derivative of a material property ```F``` with respect to a variable ```c``` would be named ```dF/dc```.
+
+To enforce this naming scheme we provide the ```DerivativeMaterialInterface```, a _veneer_ template that provides methods to _declare_ and _get_ material properties that are derivatives of other material properties.
+
+The ```DerivativeMaterialInterface``` is utilized by inheriting from it and supplying the original parent class as a template argument.
+
+# Examples
+
+The  can be used from Materials (to declare and get material property derivatives) or from Kernels (to get material property derivatives).
+
+## Use in a Material class
+
+### Header
+```c++
+#include "DerivativeMaterialInterface.h"
+#include "Material.h"
+
+[..]
+
+// we template the interface on 'Material'
+class MyMaterial : public DerivativeMaterialInterface<Material>
+{
+
+[..]
```

Comment:
uhm, I like this in theory, but I'd have a hard time finding an uncluttered example like the one I pasted in.